### PR TITLE
[Doc] Add improved 'ray start --resources' help

### LINKED ISSUE #34818: Ray Core.: ray start --resource not parsed correctly (Windows)
### What happened + What you expected to happen

The command line parser which parses the JSON dict on running the command **ray start** seems to be buggy on Windows.
(I checked Linux as well and everything is fine there)

After setting everything up the following command throws an error on the resource argument even though everything is correlty written and according to the documentation:

`ray start --head --resources='{"special_hardware": 1, "custom_label": 1}'`

with the error being: 
`--resources` is not a valid JSON string.
Valid values look like this: `--resources='{"CustomResource3": 1, "CustomResource2": 2}}'`

### Versions / Dependencies

Windows 10 Enterprise
Python version 3.10.10 
Ray version 2.4.0

### Reproduction script

Set up the environment from scratch (on a Windows machine!) by opening a terminal (cmd or powershell):

`conda create --name ray_latest -c conda-forge python=3.10`

activate env and install ray 

```
conda activate ray_latest 
pip install 'ray[default]==2.4'
```

Now run 
`ray start --head --resources='{"special_hardware": 1, "custom_label": 1}'`

Error message: 
`--resources` is not a valid JSON string.
Valid values look like this: `--resources='{"CustomResource3": 1, "CustomResource2": 2}}'`

(PS: The given example from the error message has an additional curly bracket at the end as well)

I tested the same behavior on Linux, and it works... so seems to be a specific Windows problem.

### Issue Severity

High: It blocks me from completing my task.

### PR DESCRIPTION
**Pull request by peytondmurray**
## Why are these changes needed?

This PR clarifies usage of `ray start --resources` and gives an example of valid usage on Windows.

## Related issue number

Closes https://github.com/ray-project/ray/issues/34818.

## Checks

- [x] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [x] I've run `scripts/format.sh` to lint the changes in this PR.
- [x] I've included any doc changes needed for https://docs.ray.io/en/master/.
    - [ ] I've added any new APIs to the API Reference. For example, if I added a 
           method in Tune, I've added it in `doc/source/tune/api/` under the 
           corresponding `.rst` file.
- [x] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [x] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

**Comment by mattip** [comment]
Maybe a triple-quoted string could be used to preserve line formatting, and with a touch of dedent to make it pretty? Also, note this is for the windows cmd command prompt. For powershell, additional escaping is needed. Maybe all this should only be added to the help text for windows, to give posix users less of an opportunity to laugh at windows users.
```
ray start --head --resources="{\""special_hardware\"":1, \""custom_label\"":1}"
```

**Review by mattip — APPROVED** [review]
LGTM. I confirmed the change only happens on windows. FWIW, the help looks like
```
 --resources TEXT                A JSON serialized dictionary mapping
                                  resource name to resource quantity.
                                  
                                  Windows command prompt users must ensure to
                                  double quote command line arguments; since
                                  JSON requires the use of double quotes these
                                  must be escaped as well, e.g.
                                  
                                      ray start --head
                                      --resources="{\"special_hardware\":1,
                                      \"custom_label\":1}"
                                  
                                  Windows powershell users need additional
                                  escaping:
                                  
                                      ray start --head
                                      --resources="{\""special_hardware\"":1,
                                      \""custom_label\"":1}"

```

File: python/ray/scripts/scripts.py [hunk 1]
```diff
@@ -386,7 +387,22 @@ def debug(address):
     required=False,
     default="{}",
     type=str,
-    help="a JSON serialized dictionary mapping resource name to resource quantity",
+    help="A JSON serialized dictionary mapping resource name to resource quantity."
+    + (
+        r"""
+
+Windows command prompt users must ensure to double quote command line arguments; since
```
Comment (angelinalg) [inline python/ray/scripts/scripts.py:394]:
```suggestion
Windows command prompt users must ensure to double quote command line arguments. Because
```

**Review by angelinalg — APPROVED** [review]
Some style nits.

File: python/ray/scripts/scripts.py [hunk 2]
```diff
@@ -386,7 +387,22 @@ def debug(address):
...
+JSON requires the use of double quotes these must be escaped as well, e.g.
+
+    ray start --head --resources="{\"special_hardware\":1, \"custom_label\":1}"
```
Comment (jjyao) [inline python/ray/scripts/scripts.py:397]: We can do `ray start --head --resources='{"special_hardware": 1}'` so we don't need to escape.

Comment (peytondmurray) [reply python/ray/scripts/scripts.py:397]: Does this work for Windows as well? The example I put here is taken directly from an issue where a user tried to input bad JSON similar to this (but missing the escape sequences).

Comment (mattip) [reply python/ray/scripts/scripts.py:397]: Indeed the suggested single-quote does not work. Windows is weird.
```
>ray start --head --resources='{"special_hardware": 1}'
Usage: ray start [OPTIONS]
Try 'ray start --help' for help.

Error: Got unexpected extra argument (1}')
```

Comment (peytondmurray) [reply python/ray/scripts/scripts.py:397]: Thanks for checking!

**Comment by peytondmurray** [comment]
@jjyao Would you be able to merge this?

**Review by jjyao — APPROVED** [review]

**Pull request merged**